lib/sycamore/sycamore/query/execution/sycamore_executor.py

```diff
@@ -181,22 +178,24 @@ def process_node(self, logical_node: LogicalOperator, query_id: str) -> Any:
                 logical_node=logical_node,
                 query_id=query_id,
                 inputs=inputs,
-                trace_dir=self.trace_dir,
                 s3_cache_path=s3_cache_path,
             )
         elif isinstance(logical_node, Math):
             operation = MathOperator(logical_node=logical_node, query_id=query_id, inputs=inputs)
         else:
-            raise Exception(f"Unsupported node type: {str(logical_node)}")
-
-        if result is None:
-            if self.dry_run:
-                code, imports = operation.script()
-                self.imports += imports
-                self.node_id_to_code[logical_node.node_id] = code
-                self.node_id_to_node[logical_node.node_id] = logical_node
-            else:
-                result = operation.execute()
+            raise ValueError(f"Unsupported node type: {str(logical_node)}")
+
+        if self.dry_run:
+            code, imports = operation.script()
+            self.imports += imports
+            self.node_id_to_code[logical_node.node_id] = code
+            self.node_id_to_node[logical_node.node_id] = logical_node
+            result = None
+        else:
+            result = operation.execute()
+            if trace_dir and hasattr(result, "materialize"):
```

Comment:
do all docsets have materialize? wondering if we can get this to write out traces after a top k operation

Reply:
Yes, it is a top-level docset operation.

Reply:
Hmm, it isn't writing out traces for a docset after top k (when I checked locally)